modules/geochemistry/doc/content/modules/geochemistry/tests_and_examples/bio_zoning.md

```diff
@@ -0,0 +1,153 @@
+\begin{equation}
+\mathrm{siderite} + \mathrm{HS}^{-} \rightarrow \mathrm{mackinawite} + \mathrm{HCO}_{3}^{-}
+\end{equation}
+
+The remainder of the geochemistry is standard, except for the `source_species_rates` that encode the groundwater flow that are sent from the flow App described below:
+
+!listing modules/geochemistry/test/tests/kinetics/bio_zoning_conc.i
+
+## Flow
+
+The operator-split method is used to couple the flow with the biogeochemistry, as discussed in the [theory](theory/index.md) page.  Since porosity does not change in this simulation, and the groundwater velocity is specified, only the groundwater (not the surrounding rock) is considered.  The flow is implemented using [GeochemistryTimeDerivative](GeochemistryTimeDerivative.md) and [ConservativeAdvection](ConservativeAdvection.md) Kernels, acting on the mole number per litre of groundwater of each of the chemical components that flow through the aquifer:
+
+!listing modules/geochemistry/test/tests/kinetics/bio_zoning_flow.i block=Kernels
+
+The rate of change of each component is saved into an `AuxVariable`, for instance `rate_H2O_times_vv`.  This has units mol.yr$^{-1}$, because the `Variables` represent the number of moles per litre of groundwater of each component at each node.  To convert it to mol.yr$^{-1}$.litre$^{-1}$(groundwater), a `ParsedAux` is used, such as:
+
+!listing modules/geochemistry/test/tests/kinetics/bio_zoning_flow.i block=rate_H2O_auxk
+
+Fresh groundwater is added at the left, by boundary conditions such as:
+
+!listing modules/geochemistry/test/tests/kinetics/bio_zoning_flow.i block=inject_H2O
+
+At the end of each flow timestep, the geochemistry solver is run:
+
+!listing modules/geochemistry/test/tests/kinetics/bio_zoning_flow.i block=MultiApps
+
+with the flow rates from the flow solver acting as source terms to the geochemistry simulation:
+
+!listing modules/geochemistry/test/tests/kinetics/bio_zoning_flow.i block=changes_due_to_flow
+
+After the geochemistry solver has completed its timestep, the new concentrations are retrieved from the geochemistry model in preparation for the next flow timestep:
+
+!listing modules/geochemistry/test/tests/kinetics/bio_zoning_flow.i block=transported_moles_from_geochem
+
+
+
+
+
+
+
+
+
+## Results
+
+As seen in [bio_zoning.mov], the simulation gradual evolves to steady-state in which the aquifer is partitioned into two distinct zones: the upstream zone is inhabited by the sulfate reducers, who drive the sulfate concentration to low values by around 100$\,$km from the fresh groundwater source (which sits at $x=0$).  The downstream zone is inhabited by methanogens, who produce significant dissolved methane.
+
+!media geochemistry/bio_zoning.mp4 caption=Results of the aquifer-zoning biogeochemistry reactive transport simulation  id=bio_zoning.mov
+
+
```

Comment:
This looks so cool! I'm really impressed that you have added this capability!

Reply:
Thank you @cpgr .    Your nice words mean a lot to me.